command/resource/client/grpc-flags.go

```diff
@@ -5,54 +5,60 @@ package client
 
 import (
 	"flag"
+	"strings"
 )
 
 type GRPCFlags struct {
-	address StringValue
+	address  StringValue
+	grpcTLS  BoolValue
+	certFile StringValue
+	keyFile  StringValue
+	caFile   StringValue
+	caPath   StringValue
 }
 
-// mergeFlagsIntoGRPCConfig merges flag values into grpc config
+// MergeFlagsIntoGRPCConfig merges flag values into grpc config
 // caller has to parse the CLI args before loading them into flag values
-func (f *GRPCFlags) mergeFlagsIntoGRPCConfig(c *GRPCConfig) {
+// flag values take precedence over the environment values
+func (f *GRPCFlags) MergeFlagsIntoGRPCConfig(c *GRPCConfig) {
+	if strings.HasPrefix(strings.ToLower(f.address.String()), "https://") {
+		c.GRPCTLS = true
+	}
+	f.address.Set(removeSchemaFromGRPCAddress(f.address.String()))
 	f.address.Merge(&c.Address)
+	// won't overwrite the value if it's false
+	if *f.grpcTLS.v {
```

Comment:
What if `f.grpcTLS` is nil here?

Reply:
good catch! add check:

`if f.grpcTLS.v != nil && *f.grpcTLS.v {`